ls1intum/Artemis
# `Programming exercises`: Speed up Gradle builds on Jenkins

### Checklist
#### General


#### Changes affecting Programming Exercises
~~- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).~~
- [x] I tested **all** changes and their related features with **all** corresponding user types on a local setup (Jenkins and Gitlab).

### Motivation and Context
When only a single Gradle command is executed and then the Docker container is thrown away afterwards, there is no benefit in starting the Gradle daemon. This additional overhead can therefore be saved.

### Description
Gradle has a flag `--no-daemon` that tells it to not start it.
For the static code analysis configuration the daemon is still used, as the tests and the static code analysis are split into two Gradle commands running sequentially.

I don’t know enough about the Bamboo build plans to make the same changes there, but I suppose the same optimisation applies there as well.

### Steps for Testing

Prerequisites:
- TS2 or another Gitlab/Jenkins setup
- 1 Instructor
- 1 Newly Created Programming Exercise

1. Check that you receive a build result for the template/solution repository after creating the exercise and/or after making changes to the repositories.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
unchanged

### Screenshots
n/a
